### PR TITLE
Add More Unit Tests for Robust Noise Models

### PR DESCRIPTION
**Pull request by ProfFan**
Please see the Overleaf document for rationales.

---
This change is [Reviewable](https://reviewable.io/reviews/borglab/gtsam/235)

### CODE REVIEW

File: tests/testNonlinearOptimizer.cpp [hunk 1]
```diff
@@ -339,31 +340,199 @@ TEST_UNSAFE(NonlinearOptimizer, MoreOptimization) {
...
+  // cg_params.setVerbosity("ERROR");
+  auto cg_result = NonlinearConjugateGradientOptimizer(fg, init, cg_params).optimize();
+  fg.printErrors(cg_result);
+  auto gn_result = GaussNewtonOptimizer(fg, init).optimize();
+
+  auto lm_result = LevenbergMarquardtOptimizer(fg, init, params).optimize();
+
+  auto dl_result = DoglegOptimizer(fg, init).optimize();
+
+  EXPECT(assert_equal(expected, gn_result, 1e-1));
```
Comment (dellaert) [inline tests/testNonlinearOptimizer.cpp]: Group these with calls above

File: tests/testNonlinearOptimizer.cpp [hunk 2]
```diff
@@ -339,31 +340,199 @@ TEST_UNSAFE(NonlinearOptimizer, MoreOptimization) {
...
+
+  Values expected;
+  expected.insert(0, Point2(0,0));
+  expected.insert(1, Point2(1,1.85));
 
   LevenbergMarquardtParams params;
-  EXPECT(assert_equal(expected, GaussNewtonOptimizer(fg, init).optimize()));
-  EXPECT(assert_equal(expected, LevenbergMarquardtOptimizer(fg, init, params).optimize()));
-  EXPECT(assert_equal(expected, DoglegOptimizer(fg, init).optimize()));
+  gtsam::NonlinearConjugateGradientOptimizer::Parameters cg_params;
```
Comment (dellaert) [inline tests/testNonlinearOptimizer.cpp]: Default params don't work?

File: tests/testNonlinearOptimizer.cpp [hunk 3]
```diff
@@ -339,31 +340,199 @@ TEST_UNSAFE(NonlinearOptimizer, MoreOptimization) {
...
+  params.setRelativeErrorTol(1e-20);
+
+  gtsam::NonlinearConjugateGradientOptimizer::Parameters cg_params;
+  cg_params.setErrorTol(0);
+  cg_params.setMaxIterations(10000);
+  cg_params.setRelativeErrorTol(0);
+  cg_params.setAbsoluteErrorTol(0);
+  // cg_params.setVerbosity("ERROR");
+  auto cg_result = NonlinearConjugateGradientOptimizer(fg, init, cg_params).optimize();
+//  cg_result.print("CG: ");
```
Comment (dellaert) [inline tests/testNonlinearOptimizer.cpp]: remove printing from tests

File: tests/testNonlinearOptimizer.cpp (hunk 2 above)
Comment (ProfFan) [reply tests/testNonlinearOptimizer.cpp]: We will test and report back

**Pull request merged**